qcodes/instrument_drivers/basel/sp983c_remote.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Dict, Optional
+import numpy as np
+
+from qcodes.instrument.visa import VisaInstrument
+from qcodes.instrument.parameter import Parameter, DelegateParameter
+from qcodes.utils import validators as vals
+
+
+class SP983A(VisaInstrument):
+    """
+    A driver for Basel Preamp's (SP983C) Remote Instrument - Model SP983A.
+
+    Args:
+        name
+        address
+        input_offset_voltage: (Optional) A source input offset voltage
+            parameter. The range for input is -10 to 10 Volts and it is
+            user's responsibility to ensure this. This source parameter is
+            used to set offset voltage parameter of the preamp and the
+            source parameter should represent a voltage source that is
+            connected to the "Offset Input Volgate" connector of the SP983C.
+    """
+    def __init__(self,
+                 name: str,
+                 address: str,
+                 input_offset_voltage: Optional[Parameter] = None,
+                 **kwargs: Any) -> None:
+        super().__init__(name, address, terminator="\r\n", **kwargs)
+
+        self.connect_message()
+
+        self.add_parameter(
+            "gain",
+            label="Gain",
+            unit="V/A",
+            set_cmd=self._set_gain,
+            get_cmd=self._get_gain,
+            vals=vals.Enum(1e5, 1e6, 1e7, 1e8, 1e9),
+        )
+        self.add_parameter(
+            "fcut",
+            unit="Hz",
+            label="Filter Cut-Off Frequency",
+            get_cmd=self._get_filter,
+            get_parser=self._parse_filter_value,
+            set_cmd=self._set_filter,
+            val_mapping={30: '30', 100: '100', 300: '300', 1000: '1k',
+                         3000: '3k', 10e3: '10k', 30e3: '30k',
+                         100e3: '100k', 1e6: 'FULL'},
+        )
+        self.add_parameter(
+            "overload_status",
+            label="Overload Status",
+            set_cmd=None,
```

Comment:
if this does not have a set command, shall we disallow setting it altogether?
```suggestion
            set_cmd=False,
```